bin/ch-fromhost

```diff
@@ -152,6 +171,42 @@ queue_files () {
     IFS="$old_ifs"
 }
 
+queue_ofi_dso () {
+    old_ifs="$IFS"
+    IFS="$newline"
+    d="${dest:-$2}"
+    # Directory of OFI provider dsos (*-fi.so's).
+    if [ -d "$1" ]; then
+       debug "searching $1 for ofi providers"
+       for f in "${1}/"*.so; do
+           queue_ofi_dso "$f"
+       done
+       [ -n "$ofi_found" ] || "no ofi providers found in $1"
+    # Single OFI dso provider.
+    elif is_ofi "$1"; then
+        debug "  found $1"
+        inject_ofi="${inject_ofi:+$inject_ofi$newline}$1:$d"
+        ofi_found=yes
+        # process ofi dso link paths
+        ldds=$(ldd "$1" 2>&1 | grep lib | awk '{print $3}' | sort -u)
+        for l in $ldds; do
+            ld=$(dirname "$(readlink -f "$l")")
+            # avoid duplicates and host libfabric.so location
+            if     [ "$(echo "$ld_conf" | grep -c "$ld")" -eq 0 ] \
+                && [ "$(echo "$ld" | grep -c "libfabric.so")" -eq 0 ]; then
+                queue_add_ldconf "$ld"
+            fi
+        done
+    else
+        fatal "$1 is not, or does not contain, valid OFI dso(s)"
```

Comment:
In the directory case, you don't actually know here whether it contains DSOs. You just know (1) it's not a directory and (2) it's not a file with the DSO extension.